ADAL/src/urlprotocol/ADURLProtocol.m

```diff
@@ -59,14 +60,18 @@ + (void)registerHandler:(id)handler
 }
 
 
-+ (BOOL)registerProtocol
++ (BOOL)registerProtocol:(NSString*)endURL
 {
+    s_endURL = endURL;
```

Comment:
Check to make sure s_endURL != endURL first. Shouldn't happen in practice, but it's best to be safe.